microannotate/generator.py

```diff
@@ -233,9 +233,26 @@ def __init__(
 
         async with aiofiles.open(os.path.join(self.repo.workdir, path), "wb") as f:
             if self.tokenize_enabled:
-                await f.writelines(
-                    word.group(0) + b"\n" for word in SPLIT_WORD_REGEX.finditer(content)
-                )
+                if not path.endswith(".py"):
+                    lines = (
+                        word.group(0) + b"\n"
+                        for word in SPLIT_WORD_REGEX.finditer(content)
+                    )
+                else:
+                    # In Python files, whitespaces are meaningful when used for defining
+                    # indentation, so we should not ignore them.
+                    lines = []
+                    words = list(SPLIT_WORD_REGEX.finditer(content))
```

Comment:
Maybe you should use an other regex for this case to skip starting whites: "^[ \t]*"

Reply:
Good idea, I could do `^[ \t]+` (with + instead of * so I don't match empty strings), and use re.MULTILINE (otherwise I only match at the beginning of the whole text).

Reply:
Done with 9fc4d53.